docs/quickstart.rst

```diff
@@ -129,6 +129,28 @@ certain number of old archives::
     borg prune -v $REPOSITORY --prefix '{hostname}-' \
         --keep-daily=7 --keep-weekly=4 --keep-monthly=6
 
+Pitfalls with shell variables and environment variables
+-------------------------------------------------------
+
+This applies to all environment variables you want borg to see, not just
+``BORG_PASSPHRASE``. The short explanation is: always ``export`` your variable,
+and use single quotes if you're unsure of the details of your shell's expansion
+behavior. E.g.::
+
+    export BORG_PASSPHRASE='complicated & long';
+
+This is because ``export`` exposes variables to subprocesses, which borg may be
+one of. More on ``export`` can be found in the "ENVIRONMENT" section of the
+bash(1) man page.
+
+.. note:: ``sudo`` does not inherit your environment, so ``sudo borg ...`` will not
```

Comment:
after chatting in IRC, it seems I'm touching a gold mine of complexity with my comment on `sudo`, so perhaps the following is better?

> Beware of how `sudo` interacts with environment variables: if testing a wrapper,
> for example, you may be surprised that the following export has no effect:
> ```sh
>    export BORG_PASSPHRASE='...';
>    sudo ./yourborgwrapper.sh # still prompts for password
> ```
> See the sudo(8) man page, and your system's sudo configuration for more.